### DISCUSSION TITLE
Is the persistent volume required when using a PostgreSQL or MySQL database?

### DISCUSSION THREAD

**Discussion by noahjahn** [dani-garcia/vaultwarden#2860] [Q&A] [answered]
As the title of the discussion asks: is the persistent volume still required to be used when running the docker container in combination with a PostgreSQL or MySQL database?

How stateless is the vaultwarden docker container? With a separate database service, does running the vaultwarden docker container mean no data will be lost if the container restarts?

Examples I can think of:

* Are any uploaded files stored in the database, or are they still stored on the file system of the container?
* Are user sessions stored in the database?
* ... etc

**Comment by jjlin** [comment, marked as answer]
The persistent volume is still required. The purpose of the various files is described at https://github.com/dani-garcia/vaultwarden/wiki/Backing-up-your-vault.
